crates/ide/src/inlay_hints.rs

```diff
@@ -257,26 +257,20 @@ fn is_named_constructor(
     }?;
     let expr = match expr {
         ast::Expr::CallExpr(call) => match call.expr()? {
-            ast::Expr::PathExpr(p) => p,
+            ast::Expr::PathExpr(path) => path,
             _ => return None,
         },
+        ast::Expr::PathExpr(path) => path,
         _ => return None,
     };
     let path = expr.path()?;
 
-    // Check for tuple-struct or tuple-variant in which case we can check the last segment
-    let callable = sema.type_of_expr(&ast::Expr::PathExpr(expr))?.original.as_callable(sema.db);
-    let callable_kind = callable.map(|it| it.kind());
-    if let Some(hir::CallableKind::TupleStruct(_) | hir::CallableKind::TupleEnumVariant(_)) =
-        callable_kind
-    {
-        if let Some(ctor) = path.segment() {
-            return (ctor.to_string() == ty_name).then(|| ());
-        }
-    }
-
-    // otherwise use the qualifying segment as the constructor name
-    let qual_seg = path.qualifier()?.segment()?;
+    // If it exists, use qualifying segment as the constructor name.
+    // If not, use the last segment.
+    let qual_seg = match path.qualifier() {
+        Some(qual) => qual.segment(),
+        None => path.segment(),
+    }?;
```

Comment:
This doesn't work 100% I think
```rs
mod x { pub struct Foo; }
let _ = x::Foo;
```
This won't be hidden right now will it?

Reply:
Yes, you're right, it's not hidden. I'll have a look at it again